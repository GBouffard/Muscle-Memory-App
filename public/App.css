.App {
  text-align: center;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 60px;
}

.App-header {
  background-color: darkred;
  color: whitesmoke;
}

.App-intro {
  font-size: large;
}

@keyframes App-logo-spin {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}

.App-grid {
  display: grid;
  grid-template-columns: repeat(4, auto);
  grid-gap: .5rem; /* shorthand property for grid-column-gap and grid-row-gap */
  background-color: darkslateblue;
  margin-top: 1rem;
  padding: .5rem;
}

.App-grid > div {
  background-color: aliceblue;
  padding: .5rem 0;
}

.App-grid > div:hover {
  background-color: lightblue;
}

.App-grid > div > Link {
  text-decoration: none;
}

.bigger-grid-item {
  grid-column-start: 2; /* inclusive */
  grid-column-end: 4; /* exclusive - in columns 2 & 3 */
}