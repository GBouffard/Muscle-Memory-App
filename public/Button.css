.hideButton {
  color: deepskyblue;
  background-color: aquamarine; 
  font-weight: bold;
  font-size: 25px;
  border: 5px solid darkslateblue;
  border-radius: 5px;
}

.changeLanguageButton {
  color: slategrey;
  background-color: gold;
  font-weight: italic;
  font-size: 15px;
  border: 2px solid goldenrod;
  border-style: inset;
}

.languageContainer {
  padding: 20px 10px 10px 10px;
  font-style: italic;
  font-family: "Trebuchet MS", Helvetica, sans-serif;
  text-shadow: 1px 2px 0 rgba(255,0,180,0.5);
}

.hiddenState {
  display: none;
}