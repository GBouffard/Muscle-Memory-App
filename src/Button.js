import React from 'react'

// same as
// import React, { Component } from 'react';
// export default class Button extends Component {

export default class Button extends React.Component {
  render () {
    const {
      onClick,
      text,
      className,
      id
    } = this.props

    return (
      <button
        id={id}
        className={className}
        onClick={onClick}>
        {text}
      </button>
    )
  }
}

Button.propTypes = {
  text: React.PropTypes.string.isRequired,
  onClick: React.PropTypes.func,
  className: React.PropTypes.string,
  id: React.PropTypes.string
}
