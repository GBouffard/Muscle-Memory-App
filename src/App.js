import React, { Component } from 'react';
import logo from '../public/logo.svg';
import '../public/App.css';
import '../public/Button.css';
import Button from './Button';
import SnapShotLink from './snapshot_link';
import classNames from 'classnames';
import _ from 'lodash';

class App extends Component {
  constructor(props) { 
    super(props);
    this.onClick = _.bind(this.onClick, this);
    this.onChangeLanguageClick = _.bind(this.onChangeLanguageClick, this);
    this.state = {
      isHidden: false,
      greeting: "Hello",
      buttonText: "Frenchify"
    }
  }

  render() {
    const buttonClassName = classNames('hideButton', { 'hiddenState': this.state.isHidden });
    return (
      <div className="App">
        <div className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h2>Welcome to React</h2>
        </div>
        <p className="App-intro">
          To get started, edit <code>src/App.js</code> and save to reload.
        </p>
        <div>
          <Button
            text="HIDE ME"
            id="hideMe"
            className={buttonClassName}
            onClick={this.onClick} />
        </div>
        <div>
          <SnapShotLink 
            page={'http://www.thulium69.com/'}>
            Thulium (my band)
          </SnapShotLink>
        </div>
        <div>
          <div className="languageContainer">
            {this.state.greeting}
          </div>
          <Button
            text={this.state.buttonText}
            className="changeLanguageButton"
            onClick={this.onChangeLanguageClick} />
        </div>
      </div>
    );
  }

  onClick() {
    this.setState({ 
      isHidden: true 
    });
  }

  onChangeLanguageClick() {
    this.setState({
      greeting: this.state.greeting === "Hello" ? "Bonjour" : "Hello",
      buttonText: this.state.greeting === "Hello" ? "Englishfy" : "Frenchify"
    });
  }
}

export default App;
