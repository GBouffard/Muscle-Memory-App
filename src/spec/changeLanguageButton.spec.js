import React from 'react';
import Button from '../Button';
import App from '../App';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

describe('Change language Button', () => {
  const mockChangeLanguage = jest.fn();

  describe('Unit Test', () => {
    it('renders and matches its snapshot', () => {
      // Using renderer.create from react-test-renderer to test a snapshot.
      const component = renderer.create(
        <Button
          text="Language"
          className="whatever"
          onClick={mockChangeLanguage} />
      );
      let tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });  

    it('calls the passed-in changeLanguage function when clicked', () => {
      // Using shallow from enzyme to test a mocked function.
      const buttonWrapper = shallow(
        <Button
          text="Language"
          className="whatever"
          onClick={mockChangeLanguage} />
      );  
      buttonWrapper.simulate('click');
      expect(mockChangeLanguage).toBeCalled();
    });
  });

  describe('Functional Test - within the App test', () => {
    // Using shallow from enzyme to simulate the container (the whole app) where the button is
    const appWrapper = shallow(<App />);

    it('is present with initial state text', () => {
        const languageButtonHtml = '<button class="changeLanguageButton">Frenchify</button>';
        expect(appWrapper.html()).toContain(languageButtonHtml);
    });
  });
});
