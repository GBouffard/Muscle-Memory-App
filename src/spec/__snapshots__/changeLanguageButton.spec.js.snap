exports[`Change language Button Unit Test renders and matches its snapshot 1`] = `
<button
  className="whatever"
  id={undefined}
  onClick={[Function]}>
  Language
</button>
`;
